### ISSUE TITLE
Add to the core imagefield_tokens and views404

### ISSUE THREAD

**Issue by webmetod321** [backdrop/backdrop-issues#6626]
I think it necessary to add to the core
https://www.drupal.org/project/imagefield_tokens
(makes it easy to fill a required attribute ALT using tokens)
https://www.drupal.org/project/views404
(ban indexing of unwanted paths like /view/blablabla)

I think this functionality is required by default.
Thank you for your attention.

**Comment by klonos** [comment]
Hello @webmetod321 and thank you fr taking the time to file this request. I personally find the use case for imagefield_tokens useful, but never heard of nor needed views404 before. Having said that:
- none of these modules have been ported to Backdrop contrib yet, so it indicates that people are solving these problems in other ways
- views404 has a use count of less than 1500 at the time of writing, which is only 0.5% usage across D7 sites
- the 7.x version of imagefield_tokens has a use of ~4300, which is about 1.5% usage across D7 sites

Unfortunately, the usage does not justify something that would be in core at the moment. I would start by making a request in https://github.com/backdrop-ops/contrib/issues to get these modules ported to Backdrop first, then see if there's enough usage to justify a request to add these into core.

PS: having said the above, it wouldn't be fair to point out that views404 is only 85 lines of code.

**Comment by webmetod321** [comment]
Fine. You have your own approach to evaluating such proposals.

I'll add a little.
For a content manager, manually filling out ALT is a headache. In the 21st century, it's time to automate this. I've made enough websites with content and news, and worked as a content manager, to understand this. And I used imagefield_tokens everywhere to make the work easier for myself and my colleagues. 
It is very strange that no one uses ALT, and this can be done automatically, increasing the SEO optimization of your website.

And about views404, don’t you think it’s abnormal when on request
/view/111
/view/222
/view/333
the same page /view opens, and search robots index them as duplicate content. 
Isn't this a bug in the CMS core?

Despite the ratings of these modules, this is a convenient and necessary functionality, in my opinion.
Sorry for my English.

**Comment by argiepiano** [comment]
I agree it would be great to have these as contrib in the BAckdrop space! The typical pathway is to have a module in contrib before it makes it into core. I would encourage you to post a request in page linked by klonos.

**Comment by laryn** [comment]
> For a content manager, manually filling out ALT is a headache. In the 21st century, it's time to automate this.

Just a word of caution, ALT tags are intended to be based on the content of the image itself, not simply repeating the author, title or text that is already elsewhere on the page.

But I agree with the advice so far: it makes sense to get these into contrib first.

**Comment by klonos** [comment]
> Just a word of caution, ALT tags are intended to be based on the content of the image itself, not simply repeating the author, title or text that is already elsewhere on the page.

I second that ^^ ...this is equally annoying as some people adding dots or gibberish in the alt tags when they are required. If I was to automate this in a site, I would be looking at something like the Google's Vision AI or Azure AI Vision APIs. There's been some efforts to provide integration modules for these in Drupal:
- https://www.drupal.org/project/google_vision
- https://www.drupal.org/project/azure_cognitive_services_api
- https://www.drupal.org/project/azure_vision_api
- https://www.drupal.org/project/auto_alter

These have better chances to provide more meaningful/useful alt tag text, and their efficiency will improve with time. Depending on 3rd party (paid) services, and also having privacy concerns associated with them, these should definitely be in the contrib space though.

**Comment by webmetod321** [comment]
Most entries in the blog, in the news, are accompanied by one title image, which is also displayed in announcements, in blocks of hot news, on the main page... In this case, the ALT token is configured as “page title | site name”. This is simply about the ability to configure tokens for ALT, if necessary.

I ported these Drupal modules for Backdrop for myself, and that’s enough for me. I created this post just to attract attention and show the missing functionality.

**Comment by klonos** [comment]
Thank you for porting these modules over @webmetod321 🙏🏼 ...would you be interested to move them to the official Backdrop contrib space and be their maintainer?

**Comment by webmetod321** [comment]
My competence is not enough for a maintainer, otherwise I have already done it and would not be writing this.

**Comment by kiamlaluno** [comment]
Even if those modules were already committed in *Backdrop CMS contributed projects*, an issue like this should still be created, to make the modules part of Backdrop core.

**Comment by indigoxela** [comment]
@webmetod321 for the views 404 problem there's a solution in core without additional module: use **Global null** as contextual filter (after the desired ones, of course) and set that accordingly:

![global-null-fail](https://github.com/backdrop/backdrop-issues/assets/842066/f389765e-70e1-4cd6-92b6-98865aa061a1)

This checkbox hides in the collapsed fieldset in the filter setting dialog.

**Comment by stpaultim** [comment]
> My competence is not enough for a maintainer, otherwise I have already done it and would not be writing this.

If you don't feel comfortable maintaining these projects, one possibility would be to get the work you have done into the contrib space, but without a release and a note that they still need review/work. I'd be willing to help with that. Maybe someone else interested in these modules can use your work to issue a release in the future. By sharing your code, you might at least save them some time. 

Just something to think about, but I understand if you would rather not.
